clang/lib/Sema/SemaDeclCXX.cpp

```diff
@@ -7748,6 +7748,24 @@ bool Sema::CheckExplicitlyDefaultedSpecialMember(CXXMethodDecl *MD,
         HadError = true;
       }
     }
+    // [C++23][dcl.fct.def.default]/p2.2
```

Comment:
This requirement seems to be a general one but you've implemented it as being specific to the copy and move assignment operators; can you not have this same situation with other operators (like comparisons, etc)?

Reply:
It applies to special member functions - Which excludes comparison. and constructors can't have an explicit object parameters

Reply:
Oh! I was thinking that spaceship was also a special member function that can be defaulted, but you're right, that's... not special. Thank you!